addons/mail/models/mail_thread.py

```diff
@@ -1146,7 +1167,18 @@ def _message_route_process(self, message, message_dict, routes):
             else:
                 # if a new thread is created, parent is irrelevant
                 message_dict.pop('parent_id', None)
-                thread = ModelCtx.message_new(message_dict, custom_values)
+                # Report failure/record success of message creation except if alias is not defined (fallback model case)
+                try:
+                    thread = ModelCtx.message_new(message_dict, custom_values)
+                except:
```

Comment:
Is it not safer to catch all exceptions in order to be almost certain to report that the mail is lost ? 
The main goal was to detect referenced record that has been deleted. In that case the odoo.exceptions.MissingError is raised and psycopg2.errors.ForeignKeyViolation in the test. But if there is an error in the alias_defaults, like a wrong type for the value or a miss-typed field, we get ValueError exception. 
I have updated the code to target those 3 exceptions but it is not guarantee that there may be other exception to catch.
So I wonder if it is not better to catch them all ? What do you think ?

Reply:
Yes, it’s better to catch specific exceptions but as in this case, “ForeignKeyViolation” exception is raised by the test, not used in the project and the runbot complains about its existence. So it might be a reason to break the code at one time. For that reason,  I finally opted for catching Exception as you have suggested in your last comment.